### DISCUSSION TITLE
Long commands are not wrapping correctly

### DISCUSSION THREAD

**Discussion by rawbin-** [sourcelair/xterm.js#4436] [Q&A]
when input chars length more than configured cols(80), the left chars come out from the start of the same line

## Details
- Browser and browser version: Chrome 110.0.5481.77
- OS version:   macOS 13.2.1 (22D68)
- xterm.js version: 4.18.0

### Steps to reproduce
<img width="855" alt="image" src="https://user-images.githubusercontent.com/8169783/220270204-3f6162ea-9815-45f5-a981-f056a6e8c8ef.png">

1. connected to a container bash
2. input or paste chars more than cols config

**Comment by jerch** [comment]
Thats most likely a mismatch of the PTY size and xterm.js size. On resizing the terminal you also have to update the size of the underlying PTY.

**Comment by rawbin-** [comment]
> Thats most likely a mismatch of the PTY size and xterm.js size. On resizing the terminal you also have to update the size of the underlying PTY.

Is the fit addon works for this? https://github.com/xtermjs/xterm.js/tree/master/addons/xterm-addon-fit

**Comment by jerch** [comment]
Nope, the fit addon resizes the terminal in [rows, cols] terms into a given div container size to fill it up. These values have to be transmitted to the PTY and applied there as well.

**Comment by zundaren** [comment]
Make sure to execute term.resize(), pty windowChange

**Comment by rawbin-** [comment]
<img width="1646" alt="image" src="https://user-images.githubusercontent.com/8169783/224296722-5feb8496-940c-4eaf-b553-88d498e66a70.png">

I found it only occurse in bash not in sh

**Comment by zundaren** [comment]
> <img alt="image" width="1646" src="https://user-images.githubusercontent.com/8169783/224296722-5feb8496-940c-4eaf-b553-88d498e66a70.png">
> 
> I found it only occurse in bash not in sh

It is best to wait until the drag is over before performing resize

**Comment by jerch** [comment]
Again, you need to make sure to resize the PTY as well to the new terminal size. General workflow:
- resize xterm.js to fit your purpose (eg. with fit addon to fill a certain div container up)
- if there were changes to xterm.js cols x rows size:
- send new size to backend
- apply new size to PTY - this very much depends on your PTY bridge, eg. with node-pty there is a resize method, in C it would be something along `ioctl(master_fd, TIOCSWINSZ, &winsize_struct)`

@Tyriar Do we need a document explaining this stuff? Recently we get tons of issues in that field, to me it seems ppl are unware of crucial aspects of PTY/TTY mechanics.

**Reply by Tyriar** [reply]
Something like a FAQ might be good since we see this a lot, this might also fix itself by getting a discussion showing up in search engines? I converted this to a discussion and made the title more generic

**Reply by jerch** [reply]
FAQ sounds good - would that be something for the wiki here or is it better suited for the xterm.js site? Either way sounds good to me as long as ppl can find it.

**Reply by Tyriar** [reply]
We have this discussion now which will hopefully show up in searches, looks like it's already on the wiki: https://github.com/xtermjs/xterm.js/wiki/FAQ#characters-are-being-overwritten-when-wrapped-to-a-new-line

**Comment by ashblue** [comment]
Are there a couple lines of pseudo-code somewhere explaining the workflow? It would probably stop incoming questions.

**Comment by jerch** [comment]
@ashblue Well nope there isnt as far as i am aware, therefore my question. The main issue here is the fact, that ppl have tons of different backend solutions and pty bridges, hence it cannot be stubbed easily. xterm.js is just a frontend component and has no say in backend related stuff, this is totally up to the integrator.
We could at least do something similar to the [flow control document](http://xtermjs.org/docs/guides/flowcontrol/), which tries to explain the issue with `node-pty` as backend, but cannot give ready-to-go solutions, as the real backend is unkown.
